### ISSUE TITLE
Inf values are transformed to 0

### ISSUE THREAD

**Issue by belm0** [VictoriaMetrics/vmctl#25]
Using `vmctl prometheus`, `Inf` values in the source data are written to VM as 0.

I'm not sure where it happens.  I did confirm that the values are still `Inf` at the time of `pp.im.Input()` call in prometheus.go `do()`.

**Comment by hagen1778** [comment]
Hi @belm0!
What makes you think that such values are written as 0? Pls see the added test in https://github.com/VictoriaMetrics/vmctl/commit/2c3b5e2c167d3a817bbe1c92496760056871ffb8 to check the expected output.

**Comment by valyala** [comment]
FYI, `Inf` values will be skipped starting from the next release of victoriametrics during data ingestion due to the following issue - https://github.com/VictoriaMetrics/VictoriaMetrics/issues/752

**Comment by belm0** [comment]
> What makes you think that such values are written as 0?
It's what I observed when actually using vmctl to populate a VM single-node server.  I double and triple checked.

> Pls see the added test in 2c3b5e2 to check the expected output.
I suspect the problem only appears in an actual end-to-end scenario.

**Comment by hagen1778** [comment]
> I suspect the problem only appears in an actual end-to-end scenario.

The generated by `vmctl ` json is following:
```
{"metric":{"__name__":"vm_metric_inf_positive","instance":"localhost:8428","job":"vm"},"timestamps":[1599896312402],"values":[+Inf]}
```

But looks like [json parser](https://github.com/valyala/fastjson) ignores `+` sign while parsing such values [here](https://github.com/valyala/fastjson/blob/master/fastfloat/parse.go#L58). Not sure if this is intended or not. The VM behaviour confirms that:
```
./bin/victoria-metrics --version
victoria-metrics-20200912-074903-tags-v1.41.0-0-gba74d0c1-dirty-6f47886d
curl -X POST http://localhost:8428/api/v1/import -d '{"metric":{"__name__":"float_inf"},"timestamps":[1599896312400],"values":[+Inf]}'
curl -G 'http://localhost:8428/api/v1/export' -d 'match={__name__="float_inf"}'
{"metric":{"__name__":"float_inf"},"values":[0],"timestamps":[1599896312400]}
```

I can fix it on `vmctl` side by generating `Inf` values instead of `+Inf`. Or fastjson parser may be updated to parse such values. wdyt @valyala ?

**Comment by belm0** [comment]
the most concerning behavior in terms of being prone to bugs is that `/api/v1/import` will silently write 0 on parse error

```
// ParseBestEffort parses floating-point number s.
//
// It is equivalent to strconv.ParseFloat(s, 64), but is faster.
//
// 0 is returned if the number cannot be parsed.
```

**Comment by valyala** [comment]
> I can fix it on vmctl side by generating Inf values instead of +Inf. Or fastjson parser may be updated to parse such values. wdyt @valyala ?

`fastjson` parser has been updated to properly parse `+Inf` values in the commit https://github.com/valyala/fastjson/commit/62557e7810a57083f14ab05533a551049dd1ea4d , which has been included in [v1.5.5 tag](https://github.com/valyala/fastjson/tree/v1.5.5). `fastjson` has been updated to this tag in VictoriaMetrics - see 26fa94ba8da3b5174e9d47247dd34f1e28140784 . It will be included in the next release of VictoriaMetrics.

> the most concerning behavior in terms of being prone to bugs is that /api/v1/import will silently write 0 on parse error

Agreed. This has been addressed in the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/d8183c31244b04694dbfbcfc32c18cabc83d77d8 , which will be included in the next release of VictoriaMetrics. Now VictoriaMetrics should report errors when sample values cannot be parsed instead of silently substituting such values with 0.

**Comment by valyala** [comment]
FYI, VictoriaMetrics should properly parse `+Inf` and should properly emit errors when it cannot parse timestamps and/or values in all the [supported text-based data ingestion protocols](https://victoriametrics.github.io/#how-to-import-time-series-data) starting from [release v1.41.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
